src/ember-resource.js

```diff
@@ -819,6 +819,12 @@
 
       Ember.Resource.fetch(this, ajaxOptions)
         .done(function(json) {
+          if (self.get('isDestroying') || self.get('isDestroyed')) {
+            Ember.Resource.sendEvent(self, 'didFail');
+            result.reject();
+            self.fetched().reject();
+            return;
+          }
```

Comment:
could move this into a method in the shared mixin, but there's more duplication here than this..